index.js

```diff
@@ -50,6 +54,101 @@ class ExecutorQueue extends Executor {
         this.fuseBox.addFuse(this.redisBreaker);
     }
 
+    /**
+     * Posts a new build event to the API
+     * @method postBuildEvent
+     * @param {Object} config          Configuration
+     * @param {Object} config.pipeline Pipeline of the job
+     * @param {Object} config.job      Job object to create periodic builds for
+     * @return {Promise}
+     */
+    async postBuildEvent(config) {
+        const pipeline = config.pipeline;
+        const job = config.job;
+
+        return pipeline.admin((user) => {
+            const jwt = this.tokenGen(user, {}, pipeline.scmContext);
+
+            const options = {
+                url: '/events',
+                method: 'POST',
+                headers: {
+                    Authorization: `Bearer ${jwt}`,
+                    'Content-Type': 'application/json'
+                },
+                body: {
+                    pipelineId: pipeline.id,
+                    startFrom: job.name
+                }
+            };
+
+            return req(options, (err, response) => {
+                if (err) {
+                    return Promise.reject(err);
+                }
+
+                return Promise.resolve(response);
+            });
+        });
+    }
+
+    /**
+     * Starts a new periodic build in an executor
+     * @method _startPeriodic
+     * @param {Object}   config             Configuration
+     * @param {Object}   config.pipeline    Pipeline of the job
+     * @param {Object}   config.job         Job object to create periodic builds for
+     * @param {Function} config.tokenGen    Function to generate JWT from username, scope and scmContext
+     * @param {Boolean}  config.isUpdate    Boolean to determine if updating existing periodic build
+     * @return {Promise}
+     */
+    async _startPeriodic(config, triggerBuild = false) {
```

Comment:
It invokes from here: https://github.com/screwdriver-cd/models/pull/239/files
I think it makes more sense to be here since it's a method for executor.